### PR TITLE
migrate to target dev/1.0.0, remove unwraps

### PR DESCRIPTION
**Pull request by Charles-Schleich**
Migrated to target dev/1.0.0
Minor bump to deps: `uhlc`, `serde_json`
improved error messages and logic when handling Options of Database handle and Column Family handles

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -42,21 +42,23 @@ async-std = "=1.12.0"
...
 ] }
-zenoh-codec = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-collections = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-core = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-protocol = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-util = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-keyexpr = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh_backend_traits = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
-zenoh-plugin-trait = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "main" }
+zenoh-buffers = { git = "https://github.com/eclipse-zenoh/zenoh", branch = "dev/1.0.0" }
```
Comment (Mallets) [inline Cargo.toml:53]: In `dev/1.0.0` you should not be forced to import anything else than `zenoh` crate. Could you try to remove all crates excepts for `zenoh_backend_traits` and `zenoh-plugin-trait`?

File: src/lib.rs [hunk 2]
```diff
@@ -377,11 +390,19 @@ fn put_kv(
     // Write content and encoding+timestamp in different Column Families
     let mut batch = WriteBatch::default();
     batch.put_cf(
-        db.cf_handle(CF_PAYLOADS).unwrap(),
+        db.cf_handle(CF_PAYLOADS).ok_or(zerror!(
+            "Option for ColumFamily {CF_PAYLOADS} was None, cancel put_kv"
+        ))?,
+        &key,
+        ZBuf::from(value.payload()).contiguous(),
```
Comment (Mallets) [inline src/lib.rs:397]: Use of `ZBuf` is discouraged. If you just want to get a `Cow<[u8]>` out of *payload* you can do `value.payload().into::<Cow<[u8]>>()` which should be more explicit.

File: src/lib.rs [hunk 3]
```diff
@@ -377,11 +390,19 @@ fn put_kv(
     // Write content and encoding+timestamp in different Column Families
     let mut batch = WriteBatch::default();
     batch.put_cf(
-        db.cf_handle(CF_PAYLOADS).unwrap(),
+        db.cf_handle(CF_PAYLOADS).ok_or(zerror!(
+            "Option for ColumFamily {CF_PAYLOADS} was None, cancel put_kv"
+        ))?,
+        &key,
+        value.payload().into::<Vec<u8>>(),
```
Comment (Mallets) [inline src/lib.rs:397]:
```suggestion
        value.payload().into::<Cow<[u8]>>(),
```

File: src/lib.rs [hunk 4]
```diff
@@ -413,39 +445,57 @@ fn get_kv(db: &DB, key: Option<OwnedKeyExpr>) -> ZResult<Option<(Value, Timestam
...
         }
         (Ok(Some(payload)), Ok(None)) => {
             trace!("second ok");
             // Only the payload is present in DB!
             // Possibly legacy data. Consider as encoding as APP_OCTET_STREAM and create timestamp from now()
-            Ok(Some((
-                Value::new(payload.into()).encoding(KnownEncoding::AppOctetStream.into()),
-                new_reception_timestamp(),
-            )))
+            match std::num::NonZeroU128::new(1u128)
```
Comment (milyin) [inline src/lib.rs:474]: Looks too complex, but ok. Normally this should be the `zid` from current zenoh session, but this is addintional work to pass it

File: src/lib.rs [hunk 5]
```diff
@@ -413,39 +445,57 @@ fn get_kv(db: &DB, key: Option<OwnedKeyExpr>) -> ZResult<Option<(Value, Timestam
...
-        db.get_cf(db.cf_handle(CF_PAYLOADS).unwrap(), &key),
-        db.get_cf(db.cf_handle(CF_DATA_INFO).unwrap(), &key),
-    ) {
+
+    let (cf_payloads, cf_data_info) = match (db.cf_handle(CF_PAYLOADS), db.cf_handle(CF_DATA_INFO))
+    {
+        (None, None) => {
+            bail!("Option for ColumFamily {CF_PAYLOADS} and {CF_DATA_INFO} were both 'None'")
+        }
+        (None, Some(_)) => bail!("Option for ColumFamily and {CF_DATA_INFO} were both 'None'"),
```
Comment (milyin) [inline src/lib.rs:454]: Why "both None" while only one of them is None?

**Review by milyin — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 4 above)
Comment (Charles-Schleich) [reply src/lib.rs:474]: The previous function `new_reception_timestamp()` was doing this with a hardcoded value of `1` 
So i tried to make it match the old behaviour. I agree its too complex, how can we make this simpler ?

File: src/lib.rs (hunk 5 above)
Comment (Charles-Schleich) [reply src/lib.rs:454]: Yupp, that is on me. 
Fixing and pushing.

File: src/lib.rs (hunk 3 above)
Comment (Charles-Schleich) [reply src/lib.rs]: Implemented

File: src/lib.rs (hunk 4 above)
Comment (Mallets) [reply src/lib.rs:476]: I think it's an antipattern passing by ` NonZeroU128` to create a timestamp. You should use https://github.com/eclipse-zenoh/zenoh/blob/2fccc37bb2dbe38558d58072be63dfe01ed350bd/zenoh/src/api/time.rs#L20 and passing the `session` as parameter.

File: src/lib.rs [hunk 6]
```diff
@@ -456,28 +508,34 @@ fn encode_data_info(encoding: &Encoding, timestamp: &Timestamp, deleted: bool) -
         .write(&mut writer, deleted as u8)
         .map_err(|_| zerror!("Failed to encode data-info (deleted)"))?;
     codec
-        .write(&mut writer, encoding)
+        .write(&mut writer, encoding.to_string().as_bytes())
```
Comment (Mallets) [inline src/lib.rs:511]: You can convert the `Encoding` to `EncodingProto` to support automatic serialization. You can do the opposite for deserialization.

File: src/lib.rs (hunk 4 above)
Comment (milyin) [reply src/lib.rs:476]: Let's keep it as is for now

File: src/lib.rs (hunk 6 above)
Comment (Charles-Schleich) [reply src/lib.rs:511]: Where is `EncodingProto` ?
I can't find a Struct or trait by that name in our codebase.

Comment (Mallets) [reply src/lib.rs:511]: Actually, https://github.com/eclipse-zenoh/zenoh/pull/1158 could be leveraged

Comment (Charles-Schleich) [reply src/lib.rs]: Bump review.

File: src/lib.rs [hunk 7]
```diff
@@ -455,29 +510,43 @@ fn encode_data_info(encoding: &Encoding, timestamp: &Timestamp, deleted: bool) -
...
+    let _ = encoding_bytes
+        .writer()
+        .write(writer)
         .map_err(|_| zerror!("Failed to encode data-info (encoding)"))?;
+
     Ok(result)
 }
 
 fn decode_data_info(buf: &[u8]) -> ZResult<(Encoding, Timestamp, bool)> {
     let codec = Zenoh080::new();
```
Comment (milyin) [inline src/lib.rs:526]: This is also the place to leverage new API without resorting to internal part. The `ZBytes` exposes the writer, it should not be necessary to access internal `Zenoh080`. Maybe we are lack of examples how to use the serializing API now. So this is also good place to make things right and have it as one of such examples

File: src/lib.rs [hunk 8]
```diff
@@ -413,39 +448,59 @@ fn get_kv(db: &DB, key: Option<OwnedKeyExpr>) -> ZResult<Option<(Value, Timestam
...
+            }
         }
         (Ok(None), _) => Ok(None),
         (Err(err), _) | (_, Err(err)) => Err(rocksdb_err_to_zerr(err)),
     }
 }
 
-fn encode_data_info(encoding: &Encoding, timestamp: &Timestamp, deleted: bool) -> ZResult<Vec<u8>> {
+fn encode_data_info(encoding: Encoding, timestamp: &Timestamp, deleted: bool) -> ZResult<Vec<u8>> {
     let codec = Zenoh080::new();
```
Comment (milyin) [inline src/lib.rs:500]: This is also the place to leverage new API without resorting to internal part. As @Mallets told me, the `ZBytes` exposes the writer, it should not be necessary to access internal `Zenoh080`. Maybe we are lack of examples how to use the serializing API now. So this is also good place to make things right and have it as one of such examples.

Though it's also may be good idea to merge it as is and make right serialize/deserialize implementation in separate PR.
@Mallets , your opinion?

**Review by milyin — CHANGES_REQUESTED** [review]

**Review by Mallets — APPROVED** [review]

**Pull request merged**